overlord/snapstate/handlers.go

```diff
@@ -119,6 +119,28 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 const defaultCoreSnapName = "core"
 const defaultBaseSnapsChannel = "stable"
 
+func inFlight(st *state.State, name string) (bool, error) {
+	for _, chg := range st.Changes() {
+		if chg.Status().Ready() {
+			continue
+		}
+		for _, tc := range chg.Tasks() {
+			if tc.Kind() == "link-snap" {
```

Comment:
Removals will likely not have this task, yet it's a change in flight too.

So either we need to test another kind, or perhaps given the purpose of this function, we should rename it to installInFlight or similar.

Reply:
Thanks, thats a very good point. I changed it to check for "discard-snap" too as we need to retry if the snap in question gets removed (it will trigger a reinstall of the snap in this case which is enough of a corner case that I think we just needle to handle it correct but don't need to optimize).